observability/logger.go

```diff
@@ -0,0 +1,74 @@
+package observability
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+type logSink struct {
+	name   string
+	logger *log.Logger
+}
+
+var _ logr.LogSink = &logSink{}
+
+func (l *logSink) Init(info logr.RuntimeInfo) {
+}
+
+func (l logSink) Enabled(level int) bool {
+	return true
+}
+
+// otel uses the following mapping for level to our log.Level
+// 4 = Info
+// 8 = Debug
+// 1 = Warning
+func (l logSink) Info(level int, msg string, kvs ...interface{}) {
+	var logLevel log.Level
+	switch level {
+	case 4:
+		logLevel = log.Info
+	case 8:
+		logLevel = log.Debug
+	case 1:
+		logLevel = log.Warn
+	default:
+		logLevel = log.Trace
+	}
+
+	l.logger.Logf(logLevel, "%s: %s", l.name, msg)
+	for i := 0; i < len(kvs); i += 2 {
+		l.logger.Logf(logLevel, "%s: %+v  ", kvs[i], kvs[i+1])
```

Comment:
Having these as separate log entries is not what we want - they'll be completely out of context. Just format them and append them to the main log line.